src/modules/map/components/toolbox/TimeSlider.vue

```diff
@@ -1,3 +1,305 @@
+    let yearThreshold = 10
+    if (amountOfLabelsOnScreen < 10) {
+        yearThreshold = 50
+    } else if (amountOfLabelsOnScreen < 16) {
+        yearThreshold = 25
+    }
+    return ALL_YEARS.filter((year) => year % yearThreshold === 0)
+})
+const innerBarStyle = computed(() => {
+    return { width: `${sliderWidth.value}px` }
+})
+const yearPositionOnSlider = computed(
+    () => (1 + ALL_YEARS.indexOf(currentYear.value)) * distanceBetweenLabels.value
+)
+const cursorPosition = computed(() => {
+    const yearCursorWidth = yearCursor.value?.clientWidth || 0
+    let left = yearPositionOnSlider.value - yearCursorWidth / 2
+    // we give an overlap of 12px as there is some space between the play button and the end
+    // of the slider
+    if (left > sliderWidth.value - (yearCursorWidth - 12)) {
+        left = sliderWidth.value - (yearCursorWidth - 12)
+    }
+    if (left < 0) {
+        left = 0
+    }
+    return {
+        left: `${left}px`,
+    }
+})
+const cursorArrowPosition = computed(() => {
+    return {
+        left: `${yearPositionOnSlider.value - 4.5}px`, // 4.5 is half the arrow width of 9px
+    }
+})
+const distanceBetweenLabels = computed(() => sliderWidth.value / ALL_YEARS.length)
+const innerBarStepStyle = computed(() => {
+    return {
+        width: `${distanceBetweenLabels.value}px`,
+    }
+})
+const yearsWithData = computed(() => {
+    // TODO PB-318 : alter this to also show years with partial data
+    const timeConfigs = layersWithTimestamps.value.map((layer) => layer.timeConfig)
+    let yearsInCommon = [...timeConfigs[0].years]
+    if (timeConfigs.length > 1) {
+        timeConfigs.slice(1).forEach((timeConfig) => {
+            yearsInCommon = yearsInCommon.filter((year) => timeConfig.years.includes(year))
+        })
+    }
+    return yearsInCommon
+})
+
+watch(screenWidth, (newValue) => {
+    setSliderWidth(newValue)
+})
+
+watch(layersWithTimestamps, (newLayers) => {
+    if (newLayers.length > 1) {
+        // checking if all layers can be set to the current year
+        if (
+            !newLayers
+                .map((layer) => layer.timeConfig.years.includes(currentYear.value))
+                .reduce((a, b) => a && b, true)
+        ) {
+            setCurrentYearAndDispatchToStore(
+                findMostRecentCommonYear(newLayers.map((layer) => layer.timeConfig))
+            )
+        }
+    } else {
+        // only one layer left, checking that it can comply with the current year, otherwise
+        // we take the most recent instead
+        const [onlyLayerLeft] = newLayers
+        if (!onlyLayerLeft.timeConfig.years.includes(currentYear.value)) {
+            setCurrentYearAndDispatchToStore(onlyLayerLeft.timeConfig.years[0])
+        }
+    }
+})
+
+// we can't watch currentYear and dispatch changes to the store here, otherwise the store gets
+// dispatch too many times when the user is moving the time slider (we wait for mouseup our
+// touchend to commit the change)
```

Comment:
That's good enough, but now that we have some "experience" and some utils to debounce actions, you might want to plan switching to a debounce based store dispatch approach.

Relying on both mouseup and touchend can be a bit tricky on some devices that fire both

Reply:
My goal was mostly to make a 1 to 1 translation. I'll keep that in mind when I'm developing the new features, especially with the datas that are distinct.